src/transform/src/column_knowledge.rs

```diff
@@ -485,14 +485,11 @@ pub fn optimize(
                     DatumKnowledge::from(&*e)
                 }
                 MirScalarExpr::CallVariadic { func: _, exprs } => {
-                    let mut knows = Vec::new();
-                    // We pop as many elements as the number of arguments of the CallVariadic.
-                    for _ in exprs {
-                        knows.push(knowledge_stack.pop().unwrap());
-                    }
-                    // Note that `any` is short-circuiting, so it has to be done separately from the
-                    // above popping.
-                    if knows.iter().any(|k| k.value.is_some()) {
+                    // Drain the knowledge, and reduce if any is `Some(_)`.
+                    if knowledge_stack
+                        .drain(0..exprs.len())
```

Comment:
Wouldn't this drain from the beginning, while the original code pops from the end? (And I think `knowledge_stack` might contain more than `exprs.len()` elements if the current node has left siblings.)

Reply:
Oh good point. Should be easy to fix, thanks!